src/coreComponents/linearAlgebra/interfaces/hypre/HypreVector.cpp

```diff
@@ -588,6 +588,16 @@ real64 * HypreVector::extractLocalVector()
   return hypre_VectorData( hypre_ParVectorLocalVector ( m_par_vector ) );
 }
 
+void HypreVector::extract( arrayView1d< real64 > const & localVector ) const
+{
+  GEOSX_LAI_ASSERT_EQ( localSize(), localVector.size() );
+  real64 const * const data = extractLocalVector();
+  forAll< execPolicy >( localSize(), [=] GEOSX_HOST_DEVICE ( HYPRE_Int const i )
+  {
+    localVector[i] = data[i];
+  } );
+}
```

Comment:
I don't see the use case in this PR, but I would have a preference to name this something like `extractCopy` to be clear about what is happening...not that it shouldn't be clear.

Reply:
This method is used in `SolverBase::nonlinearImplicitStep` to copy the solution out of LAI vector into an LvArray local vector. It has existed for a long time, but previously it was using the default implementation provided in `VectorBase`, which simply does a `std::copy` on the host... but it doesn't mark the data as touched on host, and things go wrong from there. I probably agree with you on the name, I was trying to make a minimal bugfix, but can also use this PR for extra cleanup.